mailbox/backup/src/test/java/org/apache/james/mailbox/backup/DefaultMailboxBackupTest.java

```diff
@@ -0,0 +1,159 @@
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.backup;
+
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Optional;
+
+import org.apache.commons.compress.archivers.zip.ZipFile;
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.blob.api.HashBlobId;
+import org.apache.james.blob.memory.MemoryBlobStore;
+import org.apache.james.core.User;
+import org.apache.james.junit.TemporaryFolderExtension;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.backup.ZipAssert.EntryChecks;
+import org.apache.james.mailbox.inmemory.MemoryMailboxManagerProvider;
+import org.apache.james.mailbox.manager.ManagerTestResources;
+import org.apache.james.mailbox.model.BlobId;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.store.mail.model.MailboxMessage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.reactivestreams.Publisher;
+import org.testcontainers.shaded.org.apache.commons.io.IOUtils;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.fge.lambdas.predicates.PredicateChainer;
+
+@ExtendWith(TemporaryFolderExtension.class)
+class DefaultMailboxBackupTest implements MailboxMessageFixture {
+
+    private static final User USER1 = User.fromUsername(ManagerTestResources.USER);
+    private static final MailboxPath MAILBOX_PATH_USER1_MAILBOX1 = MailboxPath.forUser(ManagerTestResources.USER, MAILBOX_1_NAME);
+    private static final MailboxPath MAILBOX_PATH_OTHER_USER_MAILBOX1 = MailboxPath.forUser(ManagerTestResources.OTHER_USER, MAILBOX_OTHER_USER_NAME);
+    private static final HashBlobId.Factory BLOB_ID_FACTORY = new HashBlobId.Factory();
+
+    private final Archive archive = new Zipper();
+
+    private MailboxManager mailboxManager;
+    private BlobStore store;
+    private File destination;
+    private DefaultMailboxBackup backup;
+
+    @BeforeEach
+    void beforeEach(TemporaryFolderExtension.TemporaryFolder temporaryFolder) throws Exception {
+        destination = File.createTempFile("backup-test", ".zip", temporaryFolder.getTempDir());
+        mailboxManager = MemoryMailboxManagerProvider.provideMailboxManager(new HashSet<>());
```

Comment:
What is `new HashSet` ?

Reply:
The predeleletion hooks, i wil extract it as a constant.